src/Chapter3.hs

```diff
@@ -971,6 +1117,29 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek = Monday |
+                  Tuesday |
+                  Wednesday |
+                  Thursday |
+                  Friday |
+                  Saturday |
+                  Sunday
+                  deriving (Eq,Enum,Show)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend a
+          | a == Saturday || a == Sunday = True
+          | otherwise = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂